File: carddemo-batch-python/src/carddemo_batch/models/customer.py
"""Customer model - migrated from COBOL copybook CVCUS01Y.cpy (500 bytes)."""

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from carddemo_batch.models.base import Base


class Customer(Base):
    """
    Customer entity representing the CUSTDATA VSAM file.
    
    Migrated from COBOL copybook CVCUS01Y.cpy.
    Record length: 500 bytes
    Primary key: CUST-ID (9 digits)
    """
    __tablename__ = "custdata"

    cust_id: Mapped[int] = mapped_column(primary_key=True)
    cust_first_name: Mapped[str] = mapped_column(String(25), nullable=True)
    cust_middle_name: Mapped[str] = mapped_column(String(25), nullable=True)
    cust_last_name: Mapped[str] = mapped_column(String(25), nullable=True)
    cust_addr_line_1: Mapped[str] = mapped_column(String(50), nullable=True)
    cust_addr_line_2: Mapped[str] = mapped_column(String(50), nullable=True)
    cust_addr_line_3: Mapped[str] = mapped_column(String(50), nullable=True)
    cust_addr_state_cd: Mapped[str] = mapped_column(String(2), nullable=True)
    cust_addr_country_cd: Mapped[str] = mapped_column(String(3), nullable=True)
    cust_addr_zip: Mapped[str] = mapped_column(String(10), nullable=True)
    cust_phone_num_1: Mapped[str] = mapped_column(String(15), nullable=True)
    cust_phone_num_2: Mapped[str] = mapped_column(String(15), nullable=True)
    cust_ssn: Mapped[int] = mapped_column(nullable=True)
    cust_govt_issued_id: Mapped[str] = mapped_column(String(20), nullable=True)
    cust_dob_yyyy_mm_dd: Mapped[str] = mapped_column(String(10), nullable=True)
    cust_eft_account_id: Mapped[str] = mapped_column(String(10), nullable=True)
    cust_pri_card_holder_ind: Mapped[str] = mapped_column(String(1), nullable=True)
    cust_fico_credit_score: Mapped[int] = mapped_column(nullable=True)

    def __repr__(self) -> str:
        return f"Customer(cust_id={self.cust_id}, name={self.cust_first_name} {self.cust_last_name})"

File: app-java/src/main/java/com/carddemo/model/Customer.java
package com.carddemo.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * JPA entity mapped from COBOL copybook CVCUS01Y.cpy (CUSTOMER-RECORD, RECLN 500).
 */
@Entity
@Table(name = "customers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Customer {

    /** CUST-ID PIC 9(09) */
    @Id
    @Column(name = "cust_id")
    private Long custId;

    @Column(name = "cust_first_name", length = 25)
    private String custFirstName;

    @Column(name = "cust_middle_name", length = 25)
    private String custMiddleName;

    @Column(name = "cust_last_name", length = 25)
    private String custLastName;

    @Column(name = "cust_addr_line_1", length = 50)
    private String custAddrLine1;

    @Column(name = "cust_addr_line_2", length = 50)
    private String custAddrLine2;

    @Column(name = "cust_addr_line_3", length = 50)
    private String custAddrLine3;

    @Column(name = "cust_addr_state_cd", length = 2)
    private String custAddrStateCd;

    @Column(name = "cust_addr_country_cd", length = 3)
    private String custAddrCountryCd;

    @Column(name = "cust_addr_zip", length = 10)
    private String custAddrZip;

    @Column(name = "cust_phone_num_1", length = 15)
    private String custPhoneNum1;

    @Column(name = "cust_phone_num_2", length = 15)
    private String custPhoneNum2;

    /** CUST-SSN PIC 9(09) */
    @Column(name = "cust_ssn")
    private Long custSsn;

    @Column(name = "cust_govt_issued_id", length = 20)
    private String custGovtIssuedId;

    @Column(name = "cust_dob", length = 10)
    private String custDob;

    @Column(name = "cust_eft_account_id", length = 10)
    private String custEftAccountId;

    @Column(name = "cust_pri_card_holder_ind", length = 1)
    private String custPriCardHolderInd;

    /** CUST-FICO-CREDIT-SCORE PIC 9(03) */
    @Column(name = "cust_fico_credit_score")
    private Integer custFicoCreditScore;
}
